libyul/AsmJsonConverter.cpp

```diff
@@ -66,7 +67,11 @@ Json::Value AsmJsonConverter::operator()(Literal const& _node) const
 		break;
 	}
 	ret["type"] = _node.type.str();
-	ret["value"] = _node.value.str();
+	if (util::validateUTF8(_node.value.str()))
+		ret["value"] = _node.value.str();
+	else
+		ret["value"] = Json::nullValue;
+	ret["hexValue"] = util::toHex(util::asBytes(_node.value.str()));
```

Comment:
If it's an integer literal, should the hex value go through the string conversion?

Reply:
Oh you are right, we should only have `hexValue` for string literals.